CRM/Contact/BAO/Query.php

```diff
@@ -3041,12 +3041,32 @@ public function group($values) {
     }
 
     //CRM-19589: contact(s) removed from a Smart Group, resides in civicrm_group_contact table
-    if (count($smartGroupIDs)) {
-      $groupClause[] = " ( " . $this->addGroupContactCache($smartGroupIDs, NULL, "contact_a", $op) . " ) ";
+    $groupContactCacheClause = '';
+    if (count($smartGroupIDs) || empty($value)) {
+      $isNullOp = (strpos($op, 'NULL') !== FALSE);
```

Comment:
Why are we not just doing
```
 ($op == 'IS NULL') 
```

- we don't support people playing fast & loose with the $op var do we?

Reply:
Hmm, I see this variable is not required in this function after the latest changes. Pushed the commit to remove this line completely. Thanks for identifying.